h2o-py/h2o/frame.py

```diff
@@ -414,16 +413,16 @@ def show(self, use_pandas=False):
         if self._ex is None:
             print("This H2OFrame has been removed.")
             return
-        if not self._ex._cache.is_valid(): self._frame()._ex._cache.fill()
+        if not self._ex._cache.is_valid(): self._frame()._ex._cache.fill(rows=self.nrows)
```

Comment:
Not sure if this is what we want. This would bring all rows of the frame into the python client which could simply cause OOM. We should always only fill the default number ( 10 ) or the number the user asked for as part of the `head`